### DISCUSSION TITLE
How you guys debug?

### DISCUSSION THREAD

**Discussion by enterth** [php-telegram-bot/core#1473] [Q&A]
Hi everyone,

I'm building a command to get callbackquery and manage invite (as a bot inside a channel),

I'm wondering how you guys debug on this,

My code is working outside bot manager (pure php) but i have hardtime to adapt it into functions.

Any help would be really appreciated.

**Comment by Hitmare** [comment]
I have the [logger](https://github.com/php-telegram-bot/core/blob/develop/doc/01-utils.md) active and the php error log in my webserver if the code has critical errors
